language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/codegen/ContractTypeCompanion.java

```diff
@@ -5,18 +5,22 @@
 
 import com.daml.ledger.javaapi.data.Identifier;
 
+import java.util.List;
+
 /** The commonality between {@link ContractCompanion} and {@link InterfaceCompanion}. */
 public abstract class ContractTypeCompanion<Maker, Data> {
```

Comment:
Not clear what this is, let's discuss with @S11001001

Reply:
@ray-roestenburg-da `ContractCompanion` represents templates, `InterfaceCompanion` interfaces, and the union of the two...

Reply:
(from other discussion) For an interface, the type argument to `ContractId` (`Marker`) is different from the payload type i.e. interface view (`Data`); for a template they are the same.